doc/dev/adr/0022-codegen-err-headers.md

```diff
@@ -0,0 +1,258 @@
+
+It is perfectly acceptable to use a `Field` co-typed `ErrorResult` within the same `Result/Object` as `Field`s co-typed with the new adjunct types defined herein, but this example omits the `ErrorResult` for simplicity.
+
+Note that the absence of an `ErrorResult` permits the "payloadFormat" property to specify "raw/0" or "custom/0", even though these formats normally require the Command "response" value to have a "schema" of "bytes".
+Using a "response" "schema" of `Object/Result` with no `ErrorResult` and with a `NormalResult` "schema" of "bytes" permits the mechanism of the present ADR to define header values for raw and custom-serialized responses.
+
+### ErrorCode adjunct type
+
+A `Field` that is co-typed `ErrorCode` must have a "schema" that is an `Enum` whose "valueSchema" is "string".
+This `Enum` enumerates the allowed values for the "AppErrCode" user property defined in ADR 21.
+Each "name" will be code-generated into a language-appropriate enum name, and the corresponding "enumValue" will be the string used for the "AppErrCode" header value.
+
+The `Field`'s "name" value ("appErrCode" in the above example) has no relevance to the communication, but it may affect the name of a code-generated type.
+
+The specific usage of the generated enum type will vary according to programming language.
+For the C# code described in ADR 21, the `WithAWithApplicationError` method on `ExtendedResponse` will have a code-generated form that accepts the enum type instead of a raw string for the error code.
+
+If a modeled `Command` does not include an `ErrorCode` definition, user code is expected to provide a string value directly (as illustrated in ADR 21) instead of an enumerated value.
+
+### ErrorInfo adjunct type
+
+A `Field` that is co-typed `ErrorInfo` specifies the schema for information that will be communicated via the "AppErrPayload" user property defined in ADR 21.
+In the above example, this schema is an array of strings.
+The information provided by user code will be JSON-serialized into a string for the "AppErrPayload" header value.
+
+Note that this JSON serialization is independent of the serialization format the model specifies for Command payloads.
+In the above example, the "payloadFormat" property has the value "Avro/1.11.0", indicating that AVRO serialization is used for Command (and Telemetry) payloads.
+To ensure that header values are legal UTF8 strings, JSON serialization is always used for the "AppErrPayload" value, as prescribed by ADR 19.
+
+The `Field`'s "name" value ("appErrPayload" in the above example) has no relevance to the communication, but it may affect the name of a code-generated type.
+
+The specific usage of the generated type will vary according to programming language.
+For the C# code described in ADR 21, the `WithAWithApplicationError` method on `ExtendedResponse` will have a code-generated form that accepts the generated type instead of a serialized JSON string for the error info.
```

Comment:
```suggestion
For the C# code described in ADR 21, the `WithApplicationError` method on `ExtendedResponse` will have a code-generated form that accepts the generated type instead of a serialized JSON string for the error info.
```